src/nested_artboard.cpp

```diff
@@ -28,18 +28,29 @@ void NestedArtboard::nest(Artboard* artboard) {
     m_NestedInstance->advance(0.0f);
 }
 
+static Mat2D makeTranslate(const Artboard* artboard) {
+    return Mat2D::fromTranslate(-artboard->originX() * artboard->width(),
+                                -artboard->originY() * artboard->height());
+}
+
 void NestedArtboard::draw(Renderer* renderer) {
     if (m_NestedInstance == nullptr) {
         return;
     }
     renderer->save();
-    renderer->transform(worldTransform());
-    renderer->translate(-m_NestedInstance->originX() * m_NestedInstance->width(),
-                        -m_NestedInstance->originY() * m_NestedInstance->height());
+    renderer->transform(worldTransform() * makeTranslate(m_NestedInstance));
     m_NestedInstance->draw(renderer);
     renderer->restore();
 }
 
+Core* NestedArtboard::hitTest(const Mat2D& xform, const IAABB& area) {
+    if (m_NestedInstance == nullptr) {
+        return nullptr;
+    }
+    auto mx = xform * worldTransform() * makeTranslate(m_NestedInstance);
```

Comment:
Awesome, nice to have the separate transform here as I'm sure we'll want it for cases where we build up scenes procedurally too (on top just using it for view transform)